x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/types.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DefaultEmbeddableApi } from '@kbn/embeddable-plugin/public';
+import type { AggregateQuery } from '@kbn/es-query';
+import type {
+  HasEditCapabilities,
+  PublishesDataViews,
+  PublishesTimeRange,
+} from '@kbn/presentation-publishing';
+import type { BehaviorSubject } from 'rxjs';
+import type {
+  FieldStatisticsTableEmbeddableState,
+  FieldStatsInitializerViewType,
+  FieldStatsInitialState,
+} from '../grid_embeddable/types';
+
+export interface FieldStatsControlsApi {
+  viewType$: BehaviorSubject<FieldStatsInitializerViewType>;
+  dataViewId$: BehaviorSubject<string>;
+  query$: BehaviorSubject<AggregateQuery>;
+  showDistributions$: BehaviorSubject<boolean>;
+  updateUserInput: (update: Partial<FieldStatsInitialState>) => void;
+}
+
+export type FieldStatisticsTableEmbeddableApi =
```

Comment:
API should type should also include `PublishesDataLoading` and `PublishesBlockingError`.

Reply:
Updated here https://github.com/elastic/kibana/pull/184030/commits/2e106ba25c60fff45a853a9b50adbb7e5376f989